crates/m3/src/builder/table.rs

```diff
@@ -258,6 +258,81 @@ impl<'a, F: TowerField> TableBuilder<'a, F> {
 		)
 	}
 
+	/// Adds field exponentiation column with a fixed base
+	///
+	/// ## Parameters
+	/// - `name`: Name for the column
+	/// - `pow_bits`: The bits of exponent columns from LSB to MSB
+	/// - `base`: The base to exponentiate. The field used in exponentiation will be `FSub`
+	///
+	/// ## Preconditions
+	/// * `pow_bits.len()` must be a power of 2 and less than or equal to the width of field `FSub`
+	///
+	/// ## NOTE
+	/// * The witness generation for the return column will be done inside gkr_gpa *
+	pub fn add_static_exp<FSub>(
+		&mut self,
+		name: impl ToString,
+		pow_bits: &[Col<B1>],
+		base: F,
+	) -> Col<FSub>
+	where
+		FSub: TowerField,
+		F: ExtensionField<FSub>,
+	{
+		assert!(pow_bits.len().is_power_of_two());
```

Comment:
"In fact, gkr_exp works not only with sizes that are powers of two, so I would prefer to remove this restriction to allow for special optimizations.